src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -680,6 +690,41 @@ namespace QgsWms
     return mWmsParameters[ QgsWmsParameter::WIDTH ].toInt();
   }
 
+  QString QgsWmsParameters::srcHeight() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCHEIGHT ].toString();
+  }
+
+  QString QgsWmsParameters::srcWidth() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCWIDTH ].toString();
+  }
+
+  int QgsWmsParameters::srcHeightAsInt() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCHEIGHT ].toInt();
+  }
+
+  int QgsWmsParameters::srcWidthAsInt() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCWIDTH ].toInt();
+  }
+
+  int QgsWmsParameters::getHeightAsInt() const
+  {
+    if ( QStringList( { QStringLiteral( "GetLegendGraphic" ), QStringLiteral( "GetLegendGraphics" ) } ).contains( request() ) && srcHeightAsInt() > 0 )
```

Comment:
This looks quite inefficient - either use a static QStringList or just put the two conditions separately

Reply:
> This looks quite inefficient - either use a static QStringList or just put the two conditions separately

You're totally right @nyalldawson. But once moved within `QgsRenderer::getLegendGraphics` (cf my previous comment https://github.com/qgis/QGIS/pull/9545#pullrequestreview-215585729), this method (as well as these string comparisons) won't even be necessary (and the misused method's name will be gone too).

Reply:
Ok, thanks. I'll come up with an update regarding this soon...